backend/addcorpus/serializers.py

```diff
@@ -121,13 +121,51 @@ def to_representation(self, value):
         return render_documentation_context(content)
 
 
+class CorpusConfigurationField(serializers.SlugRelatedField):
+    '''
+    Serialise a related corpus configuration as the corpus name.
+    '''
+
+    def __init__(self, **kwargs):
+        super().__init__('corpus__name', **kwargs)
+
+    def get_queryset(self):
+        return CorpusConfiguration.objects.all()
+
+    def to_representation(self, value: CorpusConfiguration) -> str:
+        return value.corpus.name
```

Comment:
This looks overengineered to me, could it be solved with a [SlugRelatedField](https://www.django-rest-framework.org/api-guide/relations/#slugrelatedfield)?

Reply:
s in, directly use the SlugRelatedField in the serializer instead of extending it.